src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -30,27 +28,44 @@ trait AuthResource extends RestResource {
     }
   }
 
+  def doIfAuthenticatedAndAuthorized[Resource](request: HttpServletRequest,
+                                               action: AuthorizedAction[Resource],
+                                               resources: Resource*)(fn: Identity => Response): Response = {
+    doIfAuthenticated(request) { identity =>
+      doIfAuthorized(request, action, resources)(fn)(identity)
+    }
+  }
+
   def doIfAuthorized[Resource](request: HttpServletRequest,
-                               response: HttpServletResponse,
                                action: AuthorizedAction[Resource],
-                               resources: Resource*)(fn: Identity => Response): Response = {
-    def isAllowed(id: Identity) = resources.forall(authorizer.isAuthorized(id, action, _))
-    def response(fn: HttpResponse => Unit): Response = {
-      val responseFacade = new ResponseFacade
-      fn(responseFacade)
-      responseFacade.response
+                               maybeResource: Option[Resource],
+                               ifNotExists: Response)(fn: Identity => Response)(implicit identity: Identity): Response =
```

Comment:
A convenient way for maybeResource, could be to supply the resource back to the block (if it is defined): 
```scala
fn:(Resource, Identity)=>Response
```